libraries/AP_Declination/generate/generate.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+'''
+generate field tables from IGRF12
+'''
+
+import pyigrf12 as igrf
+import numpy as np
+import datetime
+import os.path
+import sys
+from math import sqrt, pi, atan2, asin
+
+if not os.path.isfile("AP_Declination.h"):
+    print("Please run this tool from the AP_Declination directory")
+    sys.exit(1)
+
+def field_to_angles(x,y,z):
+    intensity = sqrt(x**2+y**2+z**2)
+    r2d = 180.0 / pi
+    declination = r2d * atan2(y, x)
+    inclination = r2d * asin(z / intensity)
+    return [intensity, inclination, declination]
+
+isv = 0
+date = datetime.datetime.now()
+itype = 1
+alt = 0.0
+
+lat = -35.362917
```

Comment:
Is this needed? Looks like it gets set appropriately on the fly later.

Reply:
fixed